add ac-3,ec-3 and flac codecs gst caps

emutavchi opened the pull request:
Allow discovery and playback of mentioned audio codecs with MSE / EME player.

woutermeek commented:
@emutavchi we will add this one after we've upstreamed and added  https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/797

eocanha reviewed (commented):
The changes look fine. I've upstreamed them here:
- Bug [237843](https://bugs.webkit.org/show_bug.cgi?id=237843)
I'll merge them back after upstream landing.

emutavchi committed:
[GStreamer][MSE] add ac-3,ec-3 and flac codecs gst caps
https://bugs.webkit.org/show_bug.cgi?id=237843

Reviewed by Philippe Normand.

AC-3, EC-3 and FLAC formats should be supported on MSE and EME on those
platforms having the appropriate decoders for them.

This patch adds the appropriate mime types, factories and caps to the
set of codecs whose local support will be tested on MSE.

This patch is authored by Eugene Mutavchi <Ievgen_Mutavchi@comcast.com>
See: https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/804

* platform/graphics/gstreamer/GStreamerRegistryScanner.cpp: Add support for the codecs, factories and caps.
* platform/graphics/gstreamer/eme/WebKitThunderDecryptorGStreamer.cpp: Add support for the caps.

git-svn-id: http://svn.webkit.org/repository/webkit/trunk@291283 268f45cc-cd09-0410-ab3c-d52691b4dbfc

eocanha commented:
I had to revert the commit and push it again on wpe-2.28 because I forgot to add a local change to the merged commit. Sorry.
